### PR TITLE
Feat: Cart Page Functionalities

### PR DESCRIPTION
**Pull request by dikshadhote**
Preview: https://dreamstorecart.netlify.app/ 
Following functionalities have been added :
1. If No Items present in cart , CTA to product page
image

2. Display Card with price details
3. Show number items added in a cart in Navbar
image

4. Dynamically change price details according to quantity of items added  (increase/decrease quantity)
image

5. Remove item from cart

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,6 +1,7 @@
 @import "https://dreamui-demo.netlify.app/dreamui-v1.css";
 @import "./components/Home/Home.css";
 @import "./components/products/Products.css";
+@import "./components/Cart/cart.css";
```
Comment (maitrakhatri) [inline src/App.css:4]: can use barrel imports here

File: src/components/navbar/Navbar.js [hunk 2]
```diff
@@ -2,8 +2,10 @@ import React from "react";
 import "./../../App.css";
 import { Link } from "react-router-dom";
 import storeLogo from "../../assets/location.png";
-
+import { useCart } from "../../context/cart-context";
 export default function Navbar() {
+  const { stateCart } = useCart();
+  const { cart } = stateCart;
```
Comment (tusharkandpal) [inline src/components/navbar/Navbar.js:8]: you can do the same in single line as `const { stateCart : { cart } } = useCart();` in future

File: src/components/Cart/Cart.js [hunk 3]
```diff
@@ -1,5 +1,164 @@
...
+        (acc, curr) =>
+          acc + Number(curr.product.orignalPrice) * Number(curr.quantity),
+        0
+      )
+    );
+    setDiscountPrice(
+      cart.reduce(
+        (acc, curr) =>
+          acc + Number(curr.product.discountPrice) * Number(curr.quantity),
+        0
```
Comment (maitrakhatri) [inline src/components/Cart/Cart.js:23]: can abstract out these reducer functions for simplicity

File: src/App.css (hunk 1 above)
Comment (tusharkandpal) [inline src/App.css:4]: please be consistent with your `file name` (talking about case)

**Review by tusharkandpal — COMMENTED** [review]
Implemented well.!

**Review by tusharkandpal — APPROVED** [review]

**Review by maitrakhatri — APPROVED** [review]
Have left some suggestions, rest looks good.
Like your folder structure

File: src/App.css (hunk 1 above)
Comment (dikshadhote) [reply src/App.css:4]: What are barrel imports?

File: src/components/navbar/Navbar.js (hunk 2 above)
Comment (dikshadhote) [reply src/components/navbar/Navbar.js:8]: ohh wow ! Thanks for suggestion

File: src/App.css (hunk 1 above)
Comment (dikshadhote) [reply src/App.css:4]: okay

Comment (maitrakhatri) [reply src/App.css:4]: > What are barrel imports?

create a index.js file and export all the components from one single file

**Pull request merged**